src/xcode/ENA/ENA/Source/Scenes/QRScanner/Model/Parser/CheckinQRCodeParser.swift

```diff
@@ -26,34 +38,110 @@ class CheckinQRCodeParser: QRCodeParsable {
 			completion(.success(.traceLocation(traceLocation)))
 		}
 		#endif
-		
-		verificationHelper.verifyQrCode(
+
+		verifyQrCode(
 			qrCodeString: qrCode,
-			appConfigurationProvider: appConfiguration,
-			onSuccess: { [weak self] traceLocation in
+			onSuccess: { traceLocation in
 				completion(.success(.traceLocation(traceLocation)))
-				self?.verificationHelper.subscriptions.removeAll()
 			},
-			onError: { [weak self] error in
+			onError: { error in
 				completion(.failure(.checkinQrError(error)))
-				self?.verificationHelper.subscriptions.removeAll()
 			}
 		)
 	}
-	
-	// MARK: - Init
 
-	init(
-		verificationHelper: QRCodeVerificationHelper,
-		appConfiguration: AppConfigurationProviding
+	// MARK: - Internal
+
+	func verifyQrCode(
```

Comment:
just a small suggestion here, looked a bit odd.

```suggestion
	func verifyQRCode(
```